core/config.h

```diff
@@ -472,9 +472,12 @@
 #define WASM_ENABLE_GC 0
 #endif
 
-/* GC binaryen compatible mode */
-#ifndef WASM_ENABLE_GC_BINARYEN
```

Comment:
Seems we should also remove GC_BINARY in wamr-compiler/CMakeLists.txt, build-scripts/config_common.cmake and build-scripts/runtime_lib.cmake.

Reply:
Done, thanks!